sklearn/feature_selection/tests/test_feature_select.py

```diff
@@ -371,6 +371,46 @@ def test_select_heuristics_regression():
         assert_less(np.sum(support[5:] == 1), 3)
 
 
+def test_select_artificial_data_ch2():
+    # Test PR #7490 works
+    # chi2(X,y)
+    # scores = array([4., 0.71]),  pvalues = array([0.046, 0.398])
+    # for the master code
+    # the result of get_support of SelectFdr is array([False, False])
+    # by PR #7490, the result is array([True, False])
+    # SelectKBest, SelectPercentile, SelectFpr, SelectFwe also tested
+    X = np.array([[10, 20], [20, 20], [20, 30]])
+    y = np.array([[1], [0], [0]])
+    scores, pvalues = chi2(X, y)
+    assert_array_almost_equal(scores, np.array([4., 0.71428571]))
+    assert_array_almost_equal(pvalues, np.array([0.04550026, 0.39802472]))
+
+    filter_fdr = SelectFdr(chi2, alpha=0.1)
+    filter_fdr.fit(X, y).transform(X)
```

Comment:
Why do you need these calls to `transform`? Can you just remove them, making the test much more succinct?